packages/evolution-legacy/src/components/survey/Section.js

```diff
@@ -167,14 +167,13 @@ export class Section extends React.Component {
         widgetStatus.errorMessage = widgetStatus.errorMessage || serverErrorMessage;
       }
 
+      // check for joined widgets:
       let join = undefined;
-      if (widgetConfig.joinWith) {
-        const previous = this.props.widgets[i - 1];
-        const previousStatus = this.props.widgets[i - 1] ? _get(this.props.interview, `widgets.${this.props.widgets[i - 1]}`, {}) : undefined;
-        const next = this.props.widgets[i + 1];
-        const nextStatus = this.props.widgets[i + 1] ? _get(this.props.interview, `widgets.${this.props.widgets[i + 1]}`, {}) : undefined;
-
-        join = widgetConfig.joinWith === previous && previousStatus.isVisible ? 'previous' : widgetConfig.joinWith === next && nextStatus.isVisible? 'next' : undefined;
+      const nextWidgetShortname = this.props.widgets[i + 1];
+      const nextWidgetConfig = this.props.surveyContext.widgets[nextWidgetShortname];
+      const nextWidgetStatus = nextWidgetShortname ? _get(this.props.interview, `widgets.${nextWidgetShortname}`, {}) : undefined;
+      if (nextWidgetStatus && ((nextWidgetConfig.joinWith === widgetShortname && nextWidgetStatus.isVisible) || (widgetConfig.joinWith === nextWidgetShortname && nextWidgetStatus.isVisible))) {
+        join = 'next';
```

Comment:
There's only one possible choice for join: next, so you might as well use a boolean called join with next